_docs/persistentstorage.md

```diff
@@ -0,0 +1,305 @@
+# Persistent Storage Setup and usage
+
+## Common
+
+### Storage class
+
+**Storage classes**
+At this moment there are three storage classes supported
+
+| Class   | Throughput/Approx matching device        | Latency          |
+|---------|------------------------------------------|------------------|
+| beta1   | hdd                                      | TBD              |
+| beta2   | ssd                                      | TBD              |
+| beta3   | NVMe                                     | TBD              |
+| default | is alias to any of the supported classes | depends on class |
+
+**Purpose of having `default` class**
+In many cases tenants may not want/need to deal with or understand differences between storage classes. Most providers will support just one storage class and may want to ~~alias~~ map it
+
+for example
+
+**provider1** just one storage class which maps to the `default` as well
+
+| Class |         |
+|-------|---------|
+| beta1 | default |
+
+**provider2** default maps to `beta1`
+
+| Class |         |
+|-------|---------|
+| beta1 | default |
+| beta2 |         |
+
+**provider3** default maps to `beta3`
+
+| Class |         |
+|-------|---------|
+| beta1 |         |
+| beta2 |         |
+| beta3 | default |
+
+**provider4** no default class
+
+| Class |     |
+|-------|-----|
+| beta1 |     |
+| beta2 |     |
+
+
+## Tenant
+
+### SDL
+
+The SDL layout has updates needs to be used in order to utilize persistent storage. All references below are based on the following [example](#SDL-Example)
+
+#### Profiles
+
+##### Multiple storage entries
+
+**NOTE** maximum amount of volumes are limited for 2 per profile!
+
+In `profile` the storage has been upgraded to support multiple entries. Both old and new notations are supported. Both examples below are showing valid use-cases of the sdl
+
+```yaml
+storage:
+  size: 512Mi
+```
+
+```yaml
+storage:
+  - size: 512Mi
+  - name: data
+    size: 1Gi
+```
+
+##### Storage name (alias)
+
+Each entry has new field `name` used by [services](#services) to reference various storage specific parameters. It can be omitted for single value usecase and default value is set to `default`
```

Comment:
Can we drop this and just make the naming mandatory? We are trying to avoid as much "magic" behavior in the SDL as possible.